pkgs/by-name/co/coredns/package.nix

```diff
@@ -6,7 +6,7 @@
   installShellFiles,
   nixosTests,
   externalPlugins ? [ ],
-  vendorHash ? "sha256-mp+0/DQTNsgAZTnLqcQq1HVLAfKr5vUGYSZlIvM7KpE=",
+  vendorHash ? "sha256-i4eQ7LwkLTfFxRUlZ68qdtqrj2k+aWNQevyhU4Iv2lw=",
```

Comment:
Why the hash would be passed in a parameter here? What's the rationale?

Reply:
No idea. This is how it was. Can we please just merge the version update and handle any cleanup separately? This is getting thoroughly into no good deed goes unpunished territory.

Reply:
No, I would prefer to fix it here now. I'm afraid that postponing the cleanup will end up as 'we will never do it'.

Reply:
I would suggest removing that `vendorHash` from the parameter and let it live in the derivation.

Easy peasy fix.

Reply:
Actually, thinking more about this: `vendorHash` is a paramater because it will change if the `externalPlugins` list is modified, right? (I didn't verify this)

Reply:
@djds is correct: if someone specifies `externalPlugins`, they also need to specify a different `vendorHash`.